internal/node_loader.js

```diff
@@ -94,13 +110,17 @@ module.constructor._resolveFilename =
   throw error;
 }
 
+let runfilesManifest;
 if (require.main === module) {
   // Set the actual entry point in the arguments list.
   // argv[0] == node, argv[1] == entry point.
   // NB: entry_point below is replaced during the build process.
   var mainScript = process.argv[1] = 'TEMPLATED_entry_point';
   try {
-    module.constructor._load(mainScript, this, /*isMain=*/true);
+    loadRunfiles(process.env.RUNFILES).then(mf => {
+      runfilesManifest = mf;
+      module.constructor._load(mainScript, this, /*isMain=*/true);
+    });
   } catch (e) {
     console.error('failed to load main ', e.stack || e);
```

Comment:
now that the code above is async, won't this exit the catch block before anything happens?